### ISSUE TITLE
Possible to workaround API Gateway timeout limits?

### ISSUE THREAD

**Issue by mojimi** [Miserlou/Zappa#1805]
I have this code that generates a PDF synchronously and the client keeps waiting on the response being returned with the PDF, sometimes it can take several minutes.

I had previously calculated that Lambda's 15 minute limit would be enough for me, but after deploying I realized that API Gateway only wait for the lambda response up to 29 seconds, so now my whole setup is invalid.

Is there a quick way I can work around this? I don't know what to do at this point and I'm out of options and time

**Comment by jneves** [comment]
You can either change the frontend (have it ask for the file several times until it's ready) or try the ALB setup, currently in PR https://github.com/Miserlou/Zappa/pull/1730.

**Comment by mojimi** [comment]
> 
> 
> You can either change the frontend (have it ask for the file several times until it's ready) or try the ALB setup, currently in PR #1730.

Thank you, so I tried implementing a sort of polling route that the user asks for an S3 key, problem is it is raising the errors that are being catched, even though such thing does not happen in local testing.

I understand it must be related to this : https://github.com/Miserlou/Zappa#catching-exceptions but I could not understand how to apply the documented solution in my problem as the try/except is in a loop and should not be returned

```
#...
s3 = boto3.client('s3')
#...
@app.route('/polling/<key>')
def polling(key):
	start_time = time()
	obj = BytesIO()
	sleep_time = 0.500
	print('Polling for obj of key=', key)
       #Long polling for 20s
	while (time() - start_time) < 20:
		try:
			s3.download_fileobj('carusoapi', key, obj)
			s3.delete_object(Bucket='carusoapi', Key=key)
			obj.seek(0)
			return send_file(obj, as_attachment=True, attachment_filename=key)
		except Exception as e:
			sleeping_for = min(sleep_time,2)
			sleep(sleeping_for)
		sleep_time *= 1.5
	return 'not ready', 204, 'plain/text'
```

This is Flask btw

**Comment by mojimi** [comment]
Nevermind, the error is unrelated, I will open a new issue
